pkg/policies/outside/outside.go

```diff
@@ -293,12 +321,28 @@ func getUsers(ctx context.Context, r repositories, owner, repo, perm,
 	var rv []string
 	for _, u := range users {
 		if u.GetPermissions()[perm] {
-			rv = append(rv, u.GetLogin())
+			if !exemptions.isExempt(repo, u.GetLogin(), perm) {
+				rv = append(rv, u.GetLogin())
+			}
 		}
 	}
 	return rv, nil
 }
 
+func (ee OutsideExemptions) isExempt(repo, user, access string) bool {
+	for _, e := range ee {
+		if !(((e.Push || e.Admin) && access == "push") || (e.Admin && access == "admin")) {
+			continue
+		}
+		if g, err := glob.Compile(e.Repo); err == nil {
```

Comment:
However, reading the docs for this library, compiling is the more expensive operation and matching is cheaper, so It might make sense to have a custom type saved somewhere (merged config?) that includes saving the glob.Glob for each entry. In that case, keeping this as a function of that custom type makes sense.

Reply:
It would make sense, but I'm unsure about where exactly to persist the globs so that they can be used for each repo in an org because the configs used within the outside package seem to be recreated for each repo.

Reply:
Compiling once per-repo is better than repo * users. We could go with that, or stick to how it is now (not pre-optimitze) and make the function a standard function without a custom type and take `[]*OutsideExemption` as a parameter.

Reply:
Ok, I put the cache into the outside OrgConfig.